tests/unit_tests/array_unit_tests.py

```diff
@@ -294,6 +294,11 @@ def testIadd(self):
         a += b
         np.testing.assert_array_equal(a.to_numpy(), np.array([2, 4, 6, 8]))
 
+    def testIaddSelfArray(self):
```

Comment:
David, delete this test, as it may collision with PR 37.

Reply:
Don't worry about that, I will pull the changes and fix it when PR 37 is merged. Right now this branch have a pull from the branch of PR 37.